modules/GeneticsPortal.py

```diff
@@ -0,0 +1,521 @@
+import pandas as pd
+import numpy as np
+import python_jsonschema_objects as pjo
+import math
+from datetime import datetime
+from multiprocessing import Pool
+import argparse
+import gzip
+import requests
+import logging
+import functools as fn
+
+# Importing settings:
+from settings import Config
+
+class genetics_portal_evidence_generator(Config):
+    """
+    Based on the provided schema this class generates evidence string for the Genetics portal
+    derived associations.
+
+    Parameters inherited from Config class
+    """
+
+    # Source name:
+    source_id = 'ot_genetics_portal'
+
+    def __init__(self, schema_json, schema_version):
+        """
+        The init function loads the json schema into a builder object and a namespace:
+        """
+
+        # Initialize json builder based on the schema:
+        self.builder = pjo.ObjectBuilder(schema_json)
+        self.evidence_builder = self.builder.build_classes()
+        self.schema_version = schema_version
+
+    def get_evidence_string(self,row):
+
+        # Variant level information:
+        self.chromosome = row['chrom']
+        self.position = row['pos']
+        self.references_allele = row['ref']
+        self.alternative_allele = row['alt']
+        self.rs_id = row['rsid']
+        self.consequence_code = row['most_severe_gene_csq']
+        self.consequence_link = row['consequence_link']
+
+        # Association level data:
+        self.sample_size = row['sample_size']
+        self.odds_ratio = row['odds_ratio']
+        self.pval_mantissa = row['pval_mantissa']
+        self.pval_exponent = row['pval_exponent']
+        self.oddsr_ci_lower = row['oddsr_ci_lower']
+        self.oddsr_ci_upper = row['oddsr_ci_upper']
+
+        # Study level information:
+        self.study_id = row['study_id']
+        self.date_added = row['pub_date']
+        self.author = row['pub_author']
+
+        # Target level information:
+        self.ensembl_gene_id = row['gene_id']
+
+        # Disease level information:
+        self.reported_trait = row['trait_reported']
+        self.efo_id = row['efo']
+
+        # Locus to gene score:
+        self.l2g_score = row['y_proba_full_model']
+
+        ##
+        ## Applying a few checks before proceeding:
+        ##
+
```

Comment:
The following condition is used to filter out evidences where disease ID is not available. Related to issue #858